src/main/java/org/cactoos/io/TempFile.java

```diff
@@ -0,0 +1,158 @@
+
+    /**
+     * Ctor.
+     * <p>
+     * Specifies empty strings for suffix and prefix, and creates the file
+     * in the filesystem's temporary directory denoted by the system property
+     * {@code java.io.tmpdir}.
+     * @since 1.0
+     */
+    public TempFile() {
+        this("", "");
+    }
+
+    /**
+     * Ctor.
+     * <p>
+     * The temporary file will be created inside the filesystem's
+     * temporary folder (system property: {@code java.io.tmpdir}).
+     * @param prefix The temp filename's prefix
+     * @param suffix The temp filename's suffix
+     * @since 1.0
+     */
+    public TempFile(
+        final String prefix,
+        final String suffix) {
+        this(
+            () -> Paths.get(System.getProperty("java.io.tmpdir")),
+            prefix,
+            suffix
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param dir The directory in which to create the temp file
+     * @param prefix The temp filename's prefix
+     * @param suffix The temp filename's suffix
+     * @since 1.0
+     */
+    public TempFile(
+        final Scalar<Path> dir,
+        final String prefix,
+        final String suffix) {
+        this(
+            dir,
+            new TextOf(prefix),
+            new TextOf(suffix)
+        );
+    }
+
+    // @todo #63 Figure out how to handle file attributes
+    //  (java.nio.file.attribute.FileAttribute). I've left them out since
+    //  their use differs wildly across platforms. There is
+    //  PosixFilePermissions that is very easy to work with, but there is
+    //  no such equivalent for non-POSIX compliant OS's like Windows.
+    //  Perhaps users don't really need this functionality, or perhaps they
+    //  need the extra control to do stuff like share temporary files between
+    //  processes. We should at least discuss whether we should let the users
+    //  have control over the file attributes upon creating the file.
+    /**
+     * Ctor.
+     * @param dir The directory in which to create the temp file
+     * @param prefix The temp filename's prefix
+     * @param suffix The temp filename's suffix
+     * @since 1.0
+     */
+    public TempFile(
+        final Scalar<Path> dir,
+        final Text prefix,
+        final Text suffix) {
+        this(
+            new StickyScalar<>(
+                () -> Files.createTempFile(
+                    dir.value(),
+                    prefix.asString(),
+                    suffix.asString()
+                )
+            )
+        );
+    }
+
+    /**
+     * Primary ctor.
+     * @param fullpath Creates the file and returns the path to it
+     * @since 1.0
+     */
+    private TempFile(final Scalar<Path> fullpath) {
+        this.file = fullpath;
+    }
+
+    @Override
+    public Path value() throws Exception {
+        return this.file.value();
+    }
+
+    @Override
+    public void close() throws IOException {
+        Files.delete(new IoCheckedScalar<>(this).value());
```

Comment:
@llorllale Why do you delete the file on close? I guess you should just close it and it could be reopened (I am not sure about the ``Files`` api, though)